pulp_smash/tests/pulp3/pulpcore/api_v3/test_crd_artifacts.py

```diff
@@ -51,6 +54,23 @@ def test_01_create(self):
             hashlib.sha256(files['file']).hexdigest()
         )
 
+    def test_01_create_no_match_data(self):
+        """Create an artifact providing the wrong digest and size."""
+        with self.assertRaises(HTTPError):
+            data = {'sha256': utils.uuid4(),
+                    'size': randint(0, 100)}
+            self.client.post(ARTIFACTS_PATH, data=data, files=self.file)
+        for artifact in self.client.get(ARTIFACTS_PATH)['results']:
+            self.assertNotEqual(artifact['sha256'], self.sha256)
+
+    def test_01_create_with_match_data(self):
+        """Create an artifact providing the right digest and size."""
+        data = {'sha256': self.sha256,
+                'size': len(self.file['file'])}
+        artifact = self.client.post(ARTIFACTS_PATH, data=data, files=self.file)
```

Comment:
This new artifact is never cleaned up. One can verify this by applying the following diff:

```diff
diff --git a/pulp_smash/tests/pulp3/pulpcore/api_v3/test_crd_artifacts.py b/pulp_smash/tests/pulp3/pulpcore/api_v3/test_crd_artifacts.py
index 92d1c46..0544e1b 100644
--- a/pulp_smash/tests/pulp3/pulpcore/api_v3/test_crd_artifacts.py
+++ b/pulp_smash/tests/pulp3/pulpcore/api_v3/test_crd_artifacts.py
@@ -2,6 +2,7 @@
 """Tests that perform actions over artifacts."""
 import hashlib
 import unittest
+from pprint import pprint
 from random import randint
 
 from requests.exceptions import HTTPError
@@ -28,6 +29,9 @@ class ArtifactTestCase(unittest.TestCase, utils.SmokeTest):
         cls.file = {'file': utils.http_get(FILE2_URL)}
         cls.sha256 = hashlib.sha256(cls.file['file']).hexdigest()
 
+    def tearDown(self):
+        pprint(self.client.get(ARTIFACTS_PATH))
+
     def test_01_create(self):
         """Create an artifact by uploading a file.
 
```

If `test_01_create_with_match_data` is removed, then the print statements don't show any artifacts being left behind.